app/Actions/Photo/Create.php

```diff
@@ -48,15 +48,16 @@ public function __construct(?ImportMode $importMode, int $intendedOwnerId)
 	 * This method may create a new database entry or update an existing
 	 * database entry.
 	 *
-	 * @param NativeLocalFile    $sourceFile the source file
-	 * @param AbstractAlbum|null $album      the targeted parent album
+	 * @param NativeLocalFile    $sourceFile           the source file
+	 * @param int                $fileLastModifiedTime the timestamp to use if there's no creation date in Exif
+	 * @param AbstractAlbum|null $album                the targeted parent album
 	 *
 	 * @return Photo the newly created or updated photo
 	 *
 	 * @throws ModelNotFoundException
 	 * @throws LycheeException
 	 */
-	public function add(NativeLocalFile $sourceFile, ?AbstractAlbum $album = null): Photo
+	public function add(NativeLocalFile $sourceFile, int $fileLastModifiedTime, ?AbstractAlbum $album = null): Photo
```

Comment:
I may be wrong but I think the function should be defined as :
```suggestion
	public function add(NativeLocalFile $sourceFile, ?int $fileLastModifiedTime = null, ?AbstractAlbum $album = null): Photo
```
And in the first line inside the function:
```php
if ($fileLastModifiedTime === null) {
  // Retrieve the filemtime() from NativeLocalFile
}
```

Nevertheless, I would make sure that $album is actually needed to be optional. From all the call I saw in that PR, it was always provided. This leads me to think that the order of arguments should be switched:
1. `NativeLocalFile`
2. `?AbstractAlbum`
3. `int` or `?int = null`

Reply:
Do you mean $fileLastModifiedTime ? Because $album is already optional...

Reply:
yes, but I am not sure that $album is actually really optional. It seems to me that in every call add() it is already defined.

Reply:
I can try to make it mandatory, but I would do it in a different PR as it's not really related to the enhancement.

Reply:
https://github.com/LycheeOrg/Lychee/pull/1822 :)

Reply:
@ildyria I do have a question regarding #1822: didn't you just removed the default value? $album is still of type ?AbstractAlbum which is still nullable (and optional)? Sorry if this sounds dumb, but my PHP knowledge is not very strong... Thanks!

Reply:
There are no dub questions. :)
`?AbstractAlbum` is still nullable but no longer optional.
```php
// This is valid php
function method(String $arg1, ?String $arg2, ?String $arg3 = 'default'): void {
	echo $arg1;
	echo "\n";
	echo $arg2 === null ? 'NULL' : $arg2;
	echo "\n";
	echo $arg3 === null ? 'NULL' : $arg3;
};
```

You can call it like this:
```php
method('arg1', 'arg2', 'arg3');
/** This will print
 * arg1
 * arg2
 * arg3
 */

method('arg1', 'arg2', null);
/** This will print
 * arg1
 * arg2
 * NULL
 */

method('arg1', null, 'arg3');
/** This will print
 * arg1
 * NULL
 * arg3
 */
```

And because the last parameter is optional = it has a default value.
```php
method('arg1', 'arg2');
/** This will print
 * arg1
 * arg2
 * default
 */


method('arg1', null);
/** This will print
 * arg1
 * NULL
 * default
 */
```

// This is not valid:
// method('arg1');

However this is not a valid php definition:
```php
// This is valid php
function method(String $arg1, ?String $arg2 = 'defaultarg2', ?String $arg3): void {
	echo $arg1;
	echo "\n";
	echo $arg2 === null ? 'NULL' : $arg2;
	echo "\n";
	echo $arg3 === null ? 'NULL' : $arg3;
};
```
Optional parameters must be defined after mandatory ones.

Hope this clarifies it?

Reply:
Things are more clear now! Thank you for the explanation!